crates/runtime/src/backtrace.rs

```diff
@@ -0,0 +1,146 @@
+//! Backtrace object and utilities.
+
+use crate::jit_frame_registry;
+
+/// Information about backtrace frame.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct BacktraceFrame {
+    pc: usize,
+}
+
+impl Default for BacktraceFrame {
+    fn default() -> Self {
+        Self { pc: 0 }
+    }
+}
+
+impl BacktraceFrame {
+    /// Current PC or IP pointer for the frame.
+    pub fn pc(&self) -> usize {
+        self.pc
+    }
+    /// Additinal frame information.
+    pub fn tag(&self) -> Option<jit_frame_registry::JITFrameTag> {
+        jit_frame_registry::find(self.pc)
+    }
+}
+
+const BACKTRACE_LIMIT: usize = 32;
+
+/// Backtrace during WebAssembly trap.
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub struct Backtrace {
+    len: usize,
+    frames: [BacktraceFrame; BACKTRACE_LIMIT],
```

Comment:
Is there a particular reason to limit this? Could this perhaps capture a `backtrace::Backtrace` natively so we could render the full (including native frames) backtrace eventually?

Reply:
Limit is due to `Eq` trait (which is required by logic related to `RECORDED_TRAP`)

Reply:
Is `Eq` necessary here? I would naively think we don't necessarily need to support it at this level, and it'd be best to have a full-fidelity backtrace if we can